src/node_file.h

```diff
@@ -351,6 +351,8 @@ class FileHandle final : public AsyncWrap, public StreamBase {
   BaseObjectPtr<FileHandleReadWrap> current_read_;
 
   BaseObjectPtr<BindingData> binding_data_;
+
+  v8::Global<v8::Promise::Resolver> close_promise_{};
```

Comment:
This promise could easily refer to a handler which refers back to the original file handle object.

This is what internal fields are for.

Reply:
I just want to make sure I understand your suggestion, you're saying to use an internal field on the FileHandle's js obj for the promise rather than a v8::Global?

Reply:
Updated to use an internal field! :-)

Reply:
> I just want to make sure I understand your suggestion, you're saying to use an internal field on the FileHandle's js obj for the promise rather than a v8::Global?

Yes, exactly :)